File: go/grains/grains.go
package grains

import (
	"errors"
)

// Square returns the number of grains on the n-th square of a chess board, assuming that the first square holds one grain and the number of grains doubles on each successive square.
func Square(sq int) (uint64, error) {
	if sq <= 0 || sq > 64 {
		return 0, errors.New("The square number must be between 1 and 64")
	}
	return 1 << (sq - 1), nil
}

// Total returns the total number of grains on a chess board, assuming that the first square holds one grain and the number of grains doubles on each successive square.
func Total() uint64 {
	return 1<<64 - 1
}

File: typescript/grains/grains.ts
function square(n: number): number {
  if (n < 1 || n > 64) throw new Error("Input must be in the range [1, 64]");

  return 2 ** (n - 1);
}

function total(): number {
  let t = 0;
  for (let i = 1; i <= 64; i++) t += square(i);
  return t;
}

export default { square, total };
